src/hal/utils/halcmd.c

```diff
@@ -753,11 +753,12 @@ static int replace_vars(char *source_str, char *dest_str, int max_chars, char **
 		var[next_delim]='\0';
 		replacement = getenv(var);
 		if (replacement == NULL) 
-                {
-                    snprintf(info, sizeof(info), "%s", var);
-                    *detail = info;
-		    return -4;
-                }
+            {
+                size_t ret = snprintf(info, sizeof(info), "%s", var);
+                if (ret > sizeof(info)) return -7;
+                *detail = info;
+            return -4;
+            }
 		if (strlimcpy(&dp, replacement, strlen(replacement), &buf_space) <0)
```

Comment:
Your editor is reformatting code, changing leading whitespace. It makes it hard to interpret the intended change.

Reply:
Looking at the LinuxCNC coding style docs, I see the problem. I have Geany set up to indent with 4 spaces, but as far as I can tell that means that tabs get displayed with a width of 4. 
I will look in to this in more detail, but might just have to accept that files with mixed tabs and spaces will always be hard to follow on my screen.